stan/math/prim/prob/wiener_full_lpdf.hpp

```diff
@@ -0,0 +1,735 @@
+  T_partials_return abstol = 0.0;
+  T_partials_return reltol = .9 * error_bound;  // eps_rel(Integration)
+  T_partials_return abstol_wiener5 = 1e-12;     // eps_abs(wiener5)
+  T_partials_return labstol_wiener5 = log(abstol_wiener5);
+  // log(eps_abs(wiener5)
+  T_partials_return Meval = 6000;
+  T_partials_return dens = 0.0;
+  T_partials_return ld = 0.0;
+  operands_and_partials<T_y_ref, T_alpha_ref, T_delta_ref, T_beta_ref, T_t0_ref,
+                        T_sv_ref, T_sw_ref, T_st0_ref>
+      ops_partials(y_ref, alpha_ref, delta_ref, beta_ref, t0_ref, sv_ref,
+                   sw_ref, st0_ref);
+
+  // calculate density and partials
+  for (size_t i = 0; i < N; i++) {
+    // Calculate 4-parameter model without inter-trial variabilities (if
+    // sv_vec[i] == 0) or 5-parameter model with inter-trial variability in
+    // drift rate (if sv_vec[i] != 0)
+    if (sw_vec[i] == 0 && st0_vec[i] == 0) {
+      const T_partials_return y_val = y_vec.val(i);
+      const T_partials_return alpha_val = alpha_vec.val(i);
+      const T_partials_return delta_val = delta_vec.val(i);
+      const T_partials_return beta_val = beta_vec.val(i);
+      const T_partials_return t0_val = t0_vec.val(i);
+      const T_partials_return sv_val = sv_vec.val(i);
+
+      dens = internal::dwiener5(y_val - t0_val, alpha_val, delta_val, beta_val,
+                                sv_val, labstol_wiener5);
+      if (labstol_wiener5 > fabs(dens) + lerror_bound_dens - log(2))
+        dens = internal::dwiener5(y_val - t0_val, alpha_val, delta_val,
+                                  beta_val, sv_val,
+                                  fabs(dens) + lerror_bound_dens - log(2));
+      ld += dens;
+
+      // computation of derivative for t and precision check in order to give
+      // the value as deriv_t to edge1 and as -deriv_t to edge5
+      T_partials_return deriv_t
+          = internal::dtdwiener5(y_val - t0_val, alpha_val, delta_val, beta_val,
+                                 sv_val, labstol_wiener5);
+      if (labstol_wiener5 > log(fabs(deriv_t)) + dens + lerror_bound - log(2))
+        deriv_t = internal::dtdwiener5(
+            y_val - t0_val, alpha_val, delta_val, beta_val, sv_val,
+            log(fabs(deriv_t)) + dens + lerror_bound - log(4));
+
+      // computation of derivatives and precision checks
+      if (!is_constant_all<T_y>::value) {
+        ops_partials.edge1_.partials_[i] = deriv_t;
+      }
+      if (!is_constant_all<T_alpha>::value) {
+        T_partials_return deriv_a
+            = internal::dadwiener5(y_val - t0_val, alpha_val, delta_val,
+                                   beta_val, sv_val, labstol_wiener5, 0);
+        if (labstol_wiener5 > log(fabs(deriv_a)) + dens + lerror_bound - log(2))
+          deriv_a = internal::dadwiener5(
+              y_val - t0_val, alpha_val, delta_val, beta_val, sv_val,
+              log(fabs(deriv_a)) + dens + lerror_bound - log(4), 0);
```

Comment:
Any other constants like `log(4)` I think you'll want to declare at the top and just reuse through out

Reply:
> Any other constants

Ok, I will adapt the code and include math/constants for log(2) and define the other constants in the next commit! :)